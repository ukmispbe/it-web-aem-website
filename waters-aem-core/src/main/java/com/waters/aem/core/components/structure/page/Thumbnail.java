package com.waters.aem.core.components.structure.page;

import com.citytechinc.cq.component.annotations.Component;
import com.citytechinc.cq.component.annotations.DialogField;
import com.citytechinc.cq.component.annotations.widgets.PathField;
import com.day.cq.dam.api.Asset;
import com.day.cq.dam.commons.util.PrefixRenditionPicker;
import com.day.cq.wcm.foundation.Image;
import com.icfolson.aem.library.core.constants.ComponentConstants;
import com.icfolson.aem.library.models.annotations.ImageInject;
import com.waters.aem.core.constants.WatersConstants;
import org.apache.sling.api.resource.Resource;
import org.apache.sling.api.resource.ResourceResolver;
import org.apache.sling.models.annotations.DefaultInjectionStrategy;
import org.apache.sling.models.annotations.Model;

import javax.inject.Inject;

@Component(value = "Thumbnail",
    group = ComponentConstants.GROUP_HIDDEN,
    path = WatersConstants.COMPONENT_PATH_STRUCTURE,
    name = WatersConstants.COMPONENT_NAME_PAGE,
    editConfig = false,
    fileName = Thumbnail.FILE_NAME,
    touchFileName = Thumbnail.FILE_NAME)
@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
public final class Thumbnail {

    static final String FILE_NAME = "thumbnail";

    private static final String RENDITION_PREFIX = "cq5dam.thumbnail.319";

    @Inject
    private ResourceResolver resourceResolver;

    // use pathfield widget because page properties dialog hides the asset finder
    @DialogField(fieldLabel = "Thumbnail Image", name = "./thumbnailImage/" + Image.PN_REFERENCE)
    @PathField(rootPath = WatersConstants.DAM_PATH)
    @ImageInject
    private Image thumbnailImage;

    public Image getThumbnailImage() {
        return thumbnailImage;
    }

    public String getThumbnailImageRendition() {
        String thumbnailImageRendition = null;

        if (thumbnailImage != null) {
            final Resource assetResource = resourceResolver.getResource(thumbnailImage.getFileReference());

            if (assetResource != null) {
                final Asset asset = assetResource.adaptTo(Asset.class);

                thumbnailImageRendition = new PrefixRenditionPicker(RENDITION_PREFIX, true)
                    .getRendition(asset)
                    .getPath();
            }
        }

        return thumbnailImageRendition;
    }
}
