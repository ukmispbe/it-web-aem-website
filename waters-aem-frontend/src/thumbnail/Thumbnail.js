import React from "react";
import ReactSVG from "react-svg";
import PropTypes from "prop-types";
import "./thumbnailStyle.css";

const Thumbnail = ({
  thumbnailSrcURL,
  thumbnailAltText,
  thumbnailTitleText,
  thumbnailLabel,
  isVideo,
  showVideoPlayButton,
  timeStamp,
  videoIconURL,
  className,
  thumbnailLabelClassName,
  handleThumbnailMouseLeave,
  handleThumbnailMouseOver,
  handleThumbnailKeyUp,
  onThumbnailClick,
  handleThumbnailError,
}) => (
  <>
    <button
      type="button"
      className={`thumbnail-container ${className}`}
      onMouseLeave={handleThumbnailMouseLeave}
      onMouseOver={handleThumbnailMouseOver}
      onFocus={handleThumbnailMouseOver}
      onKeyUp={handleThumbnailKeyUp}
      onClick={onThumbnailClick}
    >
      <div className="thumbnail-img-inner">
        {isVideo && showVideoPlayButton && (
          <div className="thumbnail-video-overlay" />
        )}
        <img
          src={thumbnailSrcURL}
          alt={thumbnailAltText}
          title={thumbnailTitleText}
          onError={handleThumbnailError}
        />
        {timeStamp && (
          <div className="thumbnail-time-stamp-bg">
            <span className="thumbnail-time-stamp-text">{timeStamp}</span>
          </div>
        )}
        {isVideo && showVideoPlayButton && (
          <div className="thumbnail-video-icon-bg">
            <ReactSVG src={videoIconURL} />
          </div>
        )}
      </div>
    </button>
    {thumbnailLabel && (
      <div className={`thumbnail-title ${thumbnailLabelClassName}`}>
        {thumbnailLabel}
      </div>
    )}
  </>
);

Thumbnail.propTypes = {
  thumbnailSrcURL: PropTypes.string,
  thumbnailAltText: PropTypes.string,
  thumbnailTitleText: PropTypes.string,
  thumbnailLabel: PropTypes.string,
  isVideo: PropTypes.bool,
  showVideoPlayButton: PropTypes.bool,
  timeStamp: PropTypes.string,
  videoIconURL: PropTypes.string,
  className: PropTypes.string,
  thumbnailLabelClassName: PropTypes.string,
  handleThumbnailMouseLeave: PropTypes.func,
  handleThumbnailMouseOver: PropTypes.func,
  handleThumbnailMouseOver: PropTypes.func,
  handleThumbnailKeyUp: PropTypes.func,
  onThumbnailClick: PropTypes.func,
  handleThumbnailError: PropTypes.func,
};

Thumbnail.defaultProps = {
  thumbnailSrcURL: "",
  thumbnailAltText: "",
  thumbnailTitleText: "",
  thumbnailLabel: "",
  isVideo: false,
  showVideoPlayButton: true,
  timeStamp: "",
  videoIconURL: "",
  className: "",
  thumbnailLabelClassName: "",
  handleThumbnailMouseLeave: () => {},
  handleThumbnailMouseOver: () => {},
  handleThumbnailMouseOver: () => {},
  handleThumbnailKeyUp: () => {},
  onThumbnailClick: () => {},
  handleThumbnailError: () => {},
};

export default Thumbnail;
