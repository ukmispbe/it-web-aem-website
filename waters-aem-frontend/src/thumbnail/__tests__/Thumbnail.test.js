import React from "react";
import { shallow } from "enzyme";
import Thumbnail from "../Thumbnail";
const defaultConfig = {
  thumbnailURL: "",
  thumbnailAltText: "",
  thumbnailTitleText: "",
  thumbnailLabel: "",
  isVideo: false,
  showVideoPlayButton: true,
  timeStamp: "",
  videoIconURL: "",
  className: "",
  thumbnailLabelClassName: "",
  handleThumbnailMouseLeave: () => {},
  handleThumbnailMouseOver: () => {},
  handleThumbnailMouseOver: () => {},
  handleThumbnailKeyUp: () => {},
  onThumbnailClick: () => {},
  handleThumbnailError: () => {},
};

describe("<Thumbnail />", () => {
  let ThumbnailComponent = "";
  beforeEach(() => {
    ThumbnailComponent = shallow(<Thumbnail {...defaultConfig} />);
  });

  test("should render correctly", () => {
    expect(ThumbnailComponent).toMatchSnapshot();
  });

  test("should render image without label", () => {
    const props = {
      thumbnailURL:
        "http://localhost:4502/content/dam/waters/emails/innovations-otto-spe-logo.jpeg",
      thumbnailAltText: "productImage",
      thumbnailTitleText: "Innovation Acquity",
      className: "thumbnail-active",
    };
    const wrapper = shallow(<Thumbnail {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should render image with label", () => {
    const props = {
      thumbnailURL:
        "http://localhost:4502/content/dam/waters/emails/innovations-otto-spe-logo.jpeg",
      thumbnailAltText: "productImage",
      thumbnailTitleText: "Innovation Acquity",
      thumbnailLabel: "Innovation Acquity RDA Smart MS",
    };
    const wrapper = shallow(<Thumbnail {...props} />);

    expect(wrapper.find(".thumbnail-title").text()).toBe(
      "Innovation Acquity RDA Smart MS"
    );
  });

  test("should render image with video play icon", () => {
    const props = {
      thumbnailURL:
        "http://localhost:4502/content/dam/waters/emails/innovations-otto-spe-logo.jpeg",
      thumbnailAltText: "productImage",
      isVideo: true,
      videoIconURL:
        "https://dev1.waters.com/content/dam/waters/en/brand-assets/icons/play.svg",
    };
    const wrapper = shallow(<Thumbnail {...props} />);

    expect(wrapper.find("ReactSVG")).toHaveLength(1);
  });

  test("should render image with time stamp", () => {
    const props = {
      thumbnailURL:
        "http://localhost:4502/content/dam/waters/emails/innovations-otto-spe-logo.jpeg",
      thumbnailAltText: "productImage",
      isVideo: true,
      showVideoPlayButton: false,
      timeStamp: "02:34",
      videoIconURL:
        "https://dev1.waters.com/content/dam/waters/en/brand-assets/icons/play.svg",
    };
    const wrapper = shallow(<Thumbnail {...props} />);
    expect(wrapper.find("ReactSVG")).toHaveLength(0);
    expect(wrapper.find(".thumbnail-time-stamp-text").text()).toBe("02:34");
  });
});
