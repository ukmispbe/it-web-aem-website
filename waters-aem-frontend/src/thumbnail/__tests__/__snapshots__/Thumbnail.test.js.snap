// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<Thumbnail /> should render image without label 1`] = `ShallowWrapper {}`;
